lib/metrics.py

```diff
@@ -348,3 +346,77 @@ def _get_db_conn(self):
                 psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT
             )
         return self.db
+
+
+class PaidAppsMetricsEmitterThread(BaseMetricsEmitterThread):
+    @property
+    def _select_stats_to_emit(self):
+        selected_stats = []
+        if buildpackutil.i_am_primary_instance():
+            selected_stats = [
+                self._inject_database_stats,
+                self._inject_storage_stats,
+                self._inject_health,
+            ]
+        selected_stats.append(self._inject_m2ee_stats)
+        return selected_stats
+
+    def _gather_metrics(self):
+        stats = {}
+        try:
+            for inject_method in self._select_stats_to_emit:
```

Comment:
This causes a subtle but important change from the previous functionality.

Because we added the `_select_stats_to_emit` method, now, if there is an error during `_inject_m2ee_stats`, we will mark the application as unhealthy (line 381), and also log an exception.

This is different from the previous behaviour: https://github.com/mendix/cf-mendix-buildpack/pull/264/files#diff-dd78eb8c4863775411e7a54feb3946b9L104
Previously if there was an error whilst capturing m2ee metrics, we do _not_ mark the application  as unhealthy, and we do _not_ log anything (unless debug mode was on).

We absolutely need to preserve this behaviour, as it was added in response to customer support tickets, and was a major contributor to false positive alerts.

Reply:
That is true, I think I wanted to put that try exception clause inside _inject_m2ee_stats, and forgot about it. Let me fix it right away.

Reply:
Yes, ensuring that `_inject_m2ee_stats` would never throw an exception would do the trick.

Reply:
Done